src/System.Text.Encoding.CodePages/tests/EncodingCodePages.cs

```diff
@@ -2,911 +2,591 @@
-        for (int i = 0; i < charIndex; i++)
-            Assert.True(expectedResult[i] == (int)buffer[i]);
-    }
-
-    [Fact]
-    public static void TestDecodingDBCS()
-    {
-        EnsureInitialization();
+            // Currently the class EncodingInfo isn't present in corefx, but this method should check for the exhaustive list
+            // When it is, comment out this line and remove the previous foreach/assert.
+            // Assert.Equal(CrossplatformDefaultEncodings, Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
 
-        string codepageName = "shift_jis"; // 932 codepage 
+            // UTF-8 potentially isn't the default everywhere if a user changes a setting, but we can hope!
+            Assert.Equal(Encoding.UTF8, Encoding.GetEncoding(0));
 
-        byte[] shiftJapaneseBuffer = new byte[]
-        {
-            0x41, 0x42, 0x43,   // Single byte
-            0x81, 0x42,         // Double bytes
-            0xE0, 0x43,         // Double bytes
-            0x44, 0x45          // Single byte
-        };
-
-        Decoder jpDecoder = Encoding.GetEncoding(codepageName).GetDecoder();
-        const int BUFFER_SIZE = 100;
-        char[] buffer = new char[BUFFER_SIZE];
-
-        int byteIndex = 0;
-        int charIndex = 0;
+            // Add the code page provider.
+            Encoding.RegisterProvider(CodePagesEncodingProvider.Instance);
 
-        while (byteIndex < shiftJapaneseBuffer.Length)
-        {
-            int charCount = jpDecoder.GetChars(shiftJapaneseBuffer, byteIndex, 1, buffer, charIndex);
-            if (charCount > 0)
+            // Adding the code page provider should keep the originals, too.
+            foreach (var mapping in CrossplatformDefaultEncodings().Union(CodePageInfo().Select(i => Map((int)i[0], (string)i[1]))))
             {
-                charIndex += charCount;
+                Encoding encoding = Encoding.GetEncoding(mapping.Key);
+                Assert.NotNull(encoding);
+                Assert.Equal(mapping.Value, encoding.WebName);
             }
-            byteIndex++;
+            // Currently the class EncodingInfo isn't present in corefx, but this method should check for the exhaustive list
+            // When it is, comment out this line and remove the previous foreach/assert.
+            // Assert.Equal(CrossplatformDefaultEncodings().Union(CodePageInfo().Select(i => Map((int)i[0], (string)i[1])).OrderBy(i => i.Key)),
+            //               Encoding.GetEncodings().OrderBy(i => i.CodePage).Select(i => Map(i.CodePage, i.WebName)));
+
+#if !PLATFORM_UNIX
+            Encoding defaultCodePage = Encoding.GetEncoding(0);
+            Assert.True(defaultCodePage.CodePage != 0);
+            Assert.True(defaultCodePage.WebName != Encoding.UTF8.WebName);
+#else
+        // Probably Unix sticks with UTF by default?
+        Assert.Equal(Encoding.UTF8, Encoding.GetEncoding(0));
```

Comment:
No, I'm not.  I more wrote that on the basis of how the original test was (probably) failing.
Any recommendations?  Just check that we get anything back?